### PR TITLE
feat: Add ability to define securityContexts and podSecurityContext

### PR DESCRIPTION
**Pull request by mkilchhofer**
This PR adds the ability to define the securityContext on both container- and pod-level.

/cc: @toscott @mattray @Pokom

### CODE REVIEW

**Comment by mattray** [comment]
This looks good to me, my understanding is we're just putting in the support for the values to be user-supplied and aren't providing any defaults, which is fine IMO. If I get confirmation of my understanding we'll merge.

File: charts/opencost/values.yaml [hunk 1]
```diff
@@ -111,6 +124,14 @@ opencost:
       limits:
         cpu: '999m'
         memory: '1Gi'
+    # -- The security options the container should be run with
+    securityContext: {}
+      # capabilities:
+      #   drop:
+      #   - ALL
+      # readOnlyRootFilesystem: true
```
Comment (toscott) [inline charts/opencost/values.yaml:132]: Just as a note to anyone leveraging this later, the UI container crashes if readOnlyRootFilesystem is true as it needs to write to tmp space.

[emerg] 1#1: mkdir() "/tmp/proxy_temp" failed (30: Read-only file system)

**Review by toscott — APPROVED** [review]
lgtm. Thanks for another contribution @mkilchhofer

**Comment by toscott** [comment]
> This looks good to me, my understanding is we're just putting in the support for the values to be user-supplied and aren't providing any defaults, which is fine IMO. If I get confirmation of my understanding we'll merge.

Agreed. This is providing additional functionality, but does not enable any of the pod security constraints by default.

**Pull request merged**